### PR TITLE
[12.0] [MIG] project_timeline

### PR DESCRIPTION
**Pull request by Nikul-Chaudhary**
Timeline view for projects
==========================

This module extends the standard tasks and projects actions to add the timeline view to them, and also adds the needed fields in the view for handling them.

### CODE REVIEW

File: project_timeline/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# Copyright 2016-2017 Tecnativa - Pedro M. Baeza
+# Copyright 2017 Tecnativa - Carlos Dauden
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+{
+    "name": "Project timeline",
+    "summary": "Timeline view for projects",
+    "version": "12.0.1.1.0",
```
Comment (tarteo) [inline project_timeline/__manifest__.py]: Reset the version number (12.0.1.0.0)

File: project_timeline/README.rst [hunk 2]
```diff
@@ -0,0 +1,56 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
```
Comment (tarteo) [inline project_timeline/README.rst]: Use the separated readme.

**Review by tarteo — CHANGES_REQUESTED** [review]

**Comment by Nikul-Chaudhary** [comment]
@tarteo Done

**Comment by Nikul-Chaudhary** [comment]
@tarteo @pedrobaeza Could you please review it?

**Review by alexey-pelykh — CHANGES_REQUESTED** [review]
I'd started with fixing runbot issues:
```
File "/home/odoo/OCB-12.0/odoo/tools/convert.py", line 612, in _tag_record
raise Exception("Cannot update missing record %r" % xid)
odoo.tools.convert.ParseError: "Cannot update missing record 'project.project_project_3'" while parsing /home/odoo/build/OCA/project/project_timeline/demo/project_project_demo.xml:12, near
<record id="project.project_project_3" model="project.project">
<field name="date_start" eval="(DateTime.today() - timedelta(days=5)).strftime('%Y-%m-%d')"/>
<field name="date" eval="(DateTime.today() + timedelta(days=2)).strftime('%Y-%m-%d')"/>
```

**Comment by alexey-pelykh** [comment]
https://github.com/OCA/project/pull/473 this is the origin

**Comment by pedrobaeza** [comment]
Please rebase

**Comment by Nikul-Chaudhary** [comment]
> Please rebase

Done

**Comment by pedrobaeza** [comment]
Check runbot log with demo data: https://runbot3.odoo-community.org/runbot/static/build/3359770-466-1b7bc5/logs/job_20_test_all.txt

**Comment by Nikul-Chaudhary** [comment]
> Check runbot log with demo data: https://runbot3.odoo-community.org/runbot/static/build/3359770-466-1b7bc5/logs/job_20_test_all.txt

@pedrobaeza Done

**Comment by pedrobaeza** [comment]
Now Travis is red

**Comment by Nikul-Chaudhary** [comment]
> Now Travis is red

@pedrobaeza now, `Mr.Travis` is happy :green_apple:

**Comment by mgielissen** [comment]
Tested but it gives this after after opening task:

Odoo Server Error

Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 654, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 312, in _handle_exception
    raise pycompat.reraise(type(exception), exception, sys.exc_info()[2])
  File "/usr/lib/python3/dist-packages/odoo/tools/pycompat.py", line 87, in reraise
    raise value
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 696, in dispatch
    result = self._call_function(**self.params)
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 344, in _call_function
    return checked_call(self.db, args, *kwargs)
  File "/usr/lib/python3/dist-packages/odoo/service/model.py", line 97, in wrapper
    return f(dbname, args, *kwargs)
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 337, in checked_call
    result = self.endpoint(*a, **kw)
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 939, in _call_
    return self.method(*args, **kw)
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 517, in response_wrap
    response = f(*args, **kw)
  File "/usr/lib/python3/dist-packages/odoo/addons/web/controllers/main.py", line 962, in call_kw
    return self._call_kw(model, method, args, kwargs)
  File "/usr/lib/python3/dist-packages/odoo/addons/web/controllers/main.py", line 954, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/usr/lib/python3/dist-packages/odoo/api.py", line 749, in call_kw
    return _call_kw_multi(method, model, args, kwargs)
  File "/usr/lib/python3/dist-packages/odoo/api.py", line 736, in _call_kw_multi
    result = method(recs, args, *kwargs)
  File "/usr/lib/python3/dist-packages/odoo/models.py", line 5446, in onchange
    record._onchange_eval(name, field_onchange[name], result)
  File "/usr/lib/python3/dist-packages/odoo/models.py", line 5299, in _onchange_eval
    method_res = method(self)
  File "/mnt/extra-addons/project_timeline/models/project_task.py", line 19, in _onchange_user
    if old_date_start > self.date_start:
TypeError: unorderable types: bool() > datetime.datetime()

**Comment by Nikul-Chaudhary** [comment]
@mgielissen Please take a look again.

**Comment by mgielissen** [comment]
> 
> 
> @mgielissen Please take a look again.

Tasks are working now.

**Review by pedrobaeza — APPROVED** [review]

**Comment by pedrobaeza** [comment]
Counting @mgielissen's implicit approval, I merge after trying myself

**Comment by mgielissen** [comment]
There is a less file, if lessc is not installed it gives an error.

**Comment by pedrobaeza** [comment]
@mgielissen but lessc is a requirement, so I don't see the problem.

**Comment by pedrobaeza** [comment]
@Nikul-Chaudhary any way, it would be great if you convert the file to scss, as less is deprecated.

**Comment by Nikul-Chaudhary** [comment]
> @Nikul-Chaudhary any way, it would be great if you convert the file to scss, as less is deprecated.

Yes, Sure

**Comment by fietiger** [comment]
1.If there are so many Projects or tasks ,no scroll bar on right side.
2.After uninstall,App Project reports " No default view of type 'timeline' could be found !",and crash.

**Pull request merged**